### PR TITLE
Auto find the right LICENSE file ending

### PR DESCRIPTION
**Pull request by SUPERCILEX**
Hey @yshrsmz,
Most people use the `LICENSE` file name for their license instead of the longer .txt version so I think it should be the default. Heck, even you're using the shortened file name! [https://github.com/yshrsmz/LicenseAdapter/blob/master/**LICENSE**](https://github.com/yshrsmz/LicenseAdapter/blob/master/LICENSE)

In addition, this is the default for new GitHub repos so that's another reason I think `LICENSE` should be the default file name.

Examples of people using the shortened version:
[https://github.com/bumptech/glide/blob/master/**LICENSE**](https://github.com/bumptech/glide/blob/master/LICENSE)
[https://github.com/googlesamples/easypermissions/blob/master/**LICENSE**](https://github.com/googlesamples/easypermissions/blob/master/LICENSE)
[https://github.com/firebase/FirebaseUI-Android/blob/master/**LICENSE**](https://github.com/firebase/FirebaseUI-Android/blob/master/LICENSE)
[https://github.com/Triple-T/gradle-play-publisher/blob/master/**LICENSE**](https://github.com/Triple-T/gradle-play-publisher/blob/master/LICENSE)
[https://github.com/firebase/firebase-jobdispatcher-android/blob/master/**LICENSE**](https://github.com/firebase/firebase-jobdispatcher-android/blob/master/LICENSE)

### CODE REVIEW

**Comment by yshrsmz** [comment]
Sounds fair enough 😜
Could you update the example too?

**Comment by SUPERCILEX** [comment]
@yshrsmz Ok, so Jake is frustrating me; he refuses to use GitHub's default `LICENSE` name: https://github.com/square/retrofit/pull/2168.

Anyway, that's good for you because this PR has changed to automatically find the right license file name. Yay! 😄 I'm happy now because I don't have to add anoying file endings to my licenses list.

File: library/src/main/java/net/yslibrary/licenseadapter/GitHubLicenseEntry.java [hunk 1]
```diff
@@ -42,8 +43,6 @@ private static String readFile(String link) {
       while ((str = in.readLine()) != null) builder.append(str).append("\n");
       in.close();
       return builder.toString();
-    } catch (MalformedURLException e) {
```
Comment (SUPERCILEX) [inline library/src/main/java/net/yslibrary/licenseadapter/GitHubLicenseEntry.java]: `MalformedURLException` extends `IOException`.

File: library/src/main/java/net/yslibrary/licenseadapter/GitHubLicenseEntry.java [hunk 2]
```diff
@@ -67,7 +66,22 @@ private void init(String licenseName, String gitRepo, String branch, License lic
...
+          Arrays.asList(Licenses.FILE_NO_EXTENSION, Licenses.FILE_TXT, Licenses.FILE_MD);
+
+      for (String relativePath : possiblePaths) {
+        String possibleUrl = license.url.replace(Licenses.FILE_AUTO, relativePath);
+        String response = readFile(possibleUrl);
+        if (!response.equals(possibleUrl)) {
+          license.text = response;
+          license.url = possibleUrl;
+          break;
+        }
```
Comment (SUPERCILEX) [inline library/src/main/java/net/yslibrary/licenseadapter/GitHubLicenseEntry.java]: I still need to think of a failure message.

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Thu Jan 05 01:00:35 JST 2017
+#Sat Jan 28 10:18:16 PST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
```
Comment (SUPERCILEX) [inline gradle/wrapper/gradle-wrapper.properties]: Make sure to run `./gradlew wrapper` when you update Gradle or it won't import properly (what happened to me today).

File: library/src/main/java/net/yslibrary/licenseadapter/GitHubLicenseEntry.java (hunk 2 above)
Comment (SUPERCILEX) [inline library/src/main/java/net/yslibrary/licenseadapter/GitHubLicenseEntry.java]: image

Done.

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (SUPERCILEX) [inline gradle/wrapper/gradle-wrapper.properties]: Hmmm, it looks like you did that in #17. Not sure why Gradle got mad at me then.

**Comment by yshrsmz** [comment]
Wow, this is awesome!
Thank you for the contribution 👍

**Comment by SUPERCILEX** [comment]
@yshrsmz Awesome! When can I expect a new release? Thanks!

**Comment by yshrsmz** [comment]
> When can I expect a new release?

I try to release a new version in a few days, but not sure since I'm a bit busy these days.
I first thought this should be v2.0, but now this can be v1.4 :)

**Comment by SUPERCILEX** [comment]
@yshrsmz SGTM! (And yup, these aren't breaking changes anymore. Yay!)

**Comment by yshrsmz** [comment]
released as v1.4.0

**Comment by SUPERCILEX** [comment]
Thanks!!!

**Pull request merged**